lib/src/priority_queue.dart

```diff
@@ -222,19 +223,14 @@ class HeapPriorityQueue<E> implements PriorityQueue<E> {
   }
 
   @override
-  List<E> toList() {
-    var list = <E>[]
-      ..length = _length
-      ..setRange(0, _length, _queue)
-      ..sort(comparison);
-    return list;
-  }
+  List<E> toList() =>
+      List<E>.generate(_length, (i) => _queue[i]!)..sort(comparison);
```

Comment:
[optional]
```suggestion
      [for (var i = 0; i < _length; i++) _qeue[i]!]..sort(comparison);
```

Reply:
leaving as is since it is closer to the original behavior (its theoretically faster since it specifies the length)